src/main/java/teammates/ui/controller/StudentHomePageAction.java

```diff
@@ -25,32 +26,36 @@ public ActionResult execute() throws EntityDoesNotExistException {
         data = new StudentHomePageData(account);
         String recentlyJoinedCourseId = getRequestParamValue(Const.ParamsNames.CHECK_PERSISTENCE_COURSE);        
         
+        List<CourseDetailsBundle> courses = new ArrayList<CourseDetailsBundle>();
+        Map<String, Boolean> sessionSubmissionStatusMap = new HashMap<String, Boolean>();
+        
         try {
-            data.courses = logic.getCourseDetailsListForStudent(account.googleId);
-    
-            data.sessionSubmissionStatusMap = generateFeedbackSessionSubmissionStatusMap(data.courses, account.googleId);
-            CourseDetailsBundle.sortDetailedCourses(data.courses);
+            courses = logic.getCourseDetailsListForStudent(account.googleId);
+            sessionSubmissionStatusMap = generateFeedbackSessionSubmissionStatusMap(courses, account.googleId);
             
-            statusToAdmin = "studentHome Page Load<br>" + "Total courses: " + data.courses.size();
+            CourseDetailsBundle.sortDetailedCourses(courses);
             
-            boolean isDataConsistent = checkEventualConsistency(recentlyJoinedCourseId);
+            statusToAdmin = "studentHome Page Load<br>" + "Total courses: " + courses.size();
+            
+            boolean isDataConsistent = checkEventualConsistency(recentlyJoinedCourseId, courses);
```

Comment:
The method too should be named `isDataConsistent(courseId,courses)` ? May be a better name is `isCourseIncluded(courseId,courses)` because that's what the method checks.